installation/scripts/install-ory.sh

```diff
@@ -0,0 +1,158 @@
+
+echo "Helm install ORY components..."
+RELEASE_NS=ory
+RELEASE_NAME=ory-stack
+SECRET_NAME=ory-hydra-credentials
+
+kubectl create ns $RELEASE_NS || true
+
+LOCAL_PERSISTENCE=$(yq ".global.ory.hydra.persistence.postgresql.enabled" ${OVERRIDE_TEMP_ORY})
+
+# Create Secret that is referenced as 'existingSecret' under chart/ory/values.yaml
+# Secret should not be recreated if it exists, mainly during Helm updates, as it will create new random values.
+# The new random values will triggered the redeployment of the postgres db and Hydra - that breaks the deployment
+# Rotating the secrets has to be done manually; the rotation of the Hydra Secrets should be done following this guide: https://www.ory.sh/docs/hydra/self-hosted/secrets-key-rotation
+if [ ! "$(kubectl get secret $SECRET_NAME -n ${RELEASE_NS})" -a "$LOCAL_PERSISTENCE" = true ]; then
+  echo "Creating secret to be used by the Ory Helm Chart..."
+  POSTGRES_USERNAME=$(yq .global.postgresql.postgresqlUsername ${OVERRIDE_TEMP_ORY})
+  POSTGRES_PASSWORD=$(generate_random 10)
+  POSTGRES_DB=$(yq .global.postgresql.postgresqlDatabase ${OVERRIDE_TEMP_ORY})
+
+  DSN=postgres://${POSTGRES_USERNAME}:${POSTGRES_PASSWORD}@${RELEASE_NAME}-postgresql.${RELEASE_NS}.svc.cluster.local:5432/${POSTGRES_DB}?sslmode=disable\&max_conn_lifetime=10s
+
+  SYSTEM=$(generate_random 32)
+  COOKIE=$(generate_random 32)
+
+  kubectl create secret generic "$SECRET_NAME" -n "$RELEASE_NS" \
+    --from-literal=dsn="${DSN}" \
+    --from-literal=secretsSystem="${SYSTEM}" \
+    --from-literal=secretsCookie="${COOKIE}" \
+    --from-literal=postgresql-password="${POSTGRES_PASSWORD}" \
+    --dry-run=client -o yaml | kubectl apply -f -
+fi
+
+# --wait is excluded as the deployment hangs; it hangs as there is a cronjob that creates the jwks secret for Oathkeeper
+# This cronjob is triggered in the statements below
+helm upgrade --install $RELEASE_NAME -f "${OVERRIDE_TEMP_ORY}" -n $RELEASE_NS "${ROOT_PATH}"/chart/ory
+
+if [[ ! ${SKIP_JWKS_ROTATION} ]]; then
+  CRONJOB=oathkeeper-jwks-rotator
+  # CronJob creates a Secret that is needed for the successful deployment of Oathkeeper
+  kubectl set image -n $RELEASE_NS cronjob $CRONJOB keys-generator=oryd/oathkeeper:v0.38.23
+  kubectl patch cronjob -n $RELEASE_NS $CRONJOB -p '{"spec":{"schedule": "*/1 * * * *"}}'
+  until [[ $(kubectl get cronjob -n $RELEASE_NS $CRONJOB --output=jsonpath={.status.lastScheduleTime}) ]]; do
+      echo "Waiting for cronjob $CRONJOB to be scheduled"
+      sleep 3
+  done
+  kubectl patch cronjob -n $RELEASE_NS $CRONJOB -p '{"spec":{"schedule": "0 0 1 * *"}}'
+fi
+
+RESULT=0
+PIDS=""
+
+kubectl rollout status deployment $RELEASE_NAME-hydra -n $RELEASE_NS --timeout=$TIMEOUT &
+PIDS="$PIDS $!"
+
+kubectl rollout status deployment $RELEASE_NAME-oathkeeper -n $RELEASE_NS --timeout=$TIMEOUT &
+PIDS="$PIDS $!"
+
+# Wait for Ory deployment to roll out as they needs to be ready for successful compass installation
+for PID in $PIDS; do
```

Comment:
I'm not sure if I like this approach of waiting for all of the processes to complete. Wouldn't it be better if we use `kubectl` to monitor Kubernetes resources instead of monitoring the processes? For example, we can use `kubectl wait` or `kubectl rollout status deployment`.

Reply:
It used to be only with `kubectl rollout status deployment` without the processes, however, that had the undesired effect of first waiting for the first deployment to roll out for the specified timeout and then waiting the second deployment for the same timeout.

For example, we can have timeout set to 30 minutes - it waits for hydra to roll out for 30 minutes and then for oathkeeper another 30 minutes resulting in 1hr of waiting time. Using processes both waitings are initiated and then they are waited upon.